libraries/standard/mqtt/include/mqtt.h

```diff
@@ -406,6 +406,26 @@ MQTTStatus_t MQTT_ReceiveLoop( MQTTContext_t * pContext,
  */
 uint16_t MQTT_GetPacketId( MQTTContext_t * pContext );
 
+/**
+ * @brief Parses the payload of an MQTT SUBACK packet that contains the status
+ * code of subscription request made for topic filters in the equivalent
+ * subscribe packet.
+ *
+ * This function can be used to iterate over the status code
+ *
+ * @param[in] pSubackPacket The SUBACK packet whose payload is to be parsed.
+ * @param[out] pPayloadStart This is populated with the starting address
+ * of the payload (or return codes for topic filters) in the SUBACK packet.
+ * @param[out] pPayloadSize This is populated with the size of the payload
+ * in the SUBACK packet. It represents the number of topic filters whose
+ * SUBACK status is present in the packet.
+ *
+ * @return #MQTTBadParameter, or #MQTTSuccess.
+ */
+MQTTStatus_t MQTT_GetSubAckPayload( const MQTTPacketInfo_t * pSubackPacket,
```

Comment:
Nit: Maybe rename to `MQTT_GetSubackStatuses`, `MQTT_GetSubackStatusCodes` or something similar? I think it would be helpful to the customer to be clearer in what the payload actually is.

Reply:
Have renamed the API to use `StatusCodes` instead of `Payload`. I agree that the name would be helpful for customers. I had originally used payload as I felt that it was appropriate for the lightweight API which represents packet level information for its APIs.